import/dproto/intermediate.d

```diff
@@ -267,3 +291,69 @@ struct Field {
 		sink("break;\n");
 	}
 }
+
+
+struct Service {
+	this(string name) {
+		this.name = name.idup;
+	}
+	string name;
+	Options options;
+	Method[] rpc;
+
+	struct Method {
+		this(string name, string documentation, string requestType, string responseType) {
+			this.name = name.idup;
+			this.documentation = documentation.idup;
+			this.requestType = requestType.idup;
+			this.responseType = responseType.idup;
+		}
+		string name;
+		string documentation;
+		string requestType;
+		string responseType;
+		Options options;
+
+		const void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt)
+		{
+			switch(fmt.spec) {
+				case 'p':
+					sink.formattedWrite("rpc %s (%s) returns (%s)", name, requestType, responseType);
+					if(options.length > 0) {
+						sink(" {\n");
+						foreach(opt, val; options) {
+							sink.formattedWrite("option %s = %s;\n", opt, val);
+						}
+						sink("}\n");
+					} else {
+						sink(";\n");
+					}
+					break;
+				case 'i':
+					sink.formattedWrite("%s %s (%s); { %s res; return res; }\n", responseType, name, requestType);
+					break;
+				default:
+					sink.formattedWrite("%s %s (%s);\n", responseType, name, requestType);
+					break;
+			}
+		}
+	}
+
+	const void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt)
+	{
+		switch(fmt.spec) {
+			case 'p':
+				sink.formattedWrite("service %s {\n", name);
+				break;
+			case 'i':
```

Comment:
It seems like RPC stub implementation should be the default, given that it's the default for C++. The other options are handy -- I think that ".1" (implementation only) might even be a useful flag for message, only returning a dumb primitive struct, instead of an implementation.